lib/phubme/nicknames_matcher.ex

```diff
@@ -7,18 +7,19 @@ defmodule PhubMe.NicknamesMatcher do
     {:no_nicknames_found}
   end
 
-  defp matching_nicknames([ [ nickname ] | tail]) do
-    case nickname_from_mix_config(nickname) do
-      {:ok, matching_nickname} ->
-        [ [ nickname, matching_nickname ] | matching_nicknames(tail) ]
-      :error ->
-        IO.puts "No matching nickname found for " <> nickname
-        matching_nicknames(tail)
-    end
+  defp matching_nicknames(list, acc \\ [])
+
+  defp matching_nicknames([ [ nickname ] | tail], acc) do
+    next_acc =
+      case nickname_from_mix_config(nickname) do
+        {:ok, matching_nickname} -> [ [ nickname, matching_nickname ] | acc]
```

Comment:
This way of building a list is much more efficient that doing `Enum.concat`, at least in other functional programming languages. The price of this change is the `Enum.reverse` later.